readthedocs/core/static-src/core/js/doc-embed/search.js

```diff
@@ -64,6 +37,48 @@ function attach_elastic_search_query_sphinx(data) {
           }
         };
 
+        /**
+         * Build a section with its matching results.
+         *
+         * @param {String} title.
+         * @param {String} link.
+         * @param {Array} contents.
+         */
+        var buildSection = function (title, link, contents) {
+            var div_title = document.createElement("div");
+            var a_element = document.createElement("a");
+            a_element.href = link;
+            a_element.innerHTML = title;
+            div_title.appendChild(a_element);
+            html = div_title.outerHTML;
+            for (var i = 0; i < contents.length; i += 1) {
+                var div_content = document.createElement("div");
+                div_content.innerHTML = contents[i];
+                html += div_content.outerHTML;
+            }
+            return html;
+        };
+
+        /**
+         * Build a domain section.
+         *
+         * @param {String} title.
+         * @param {String} link.
+         * @param {String} content.
+         */
+        var buildDomain = function (title, link, content) {
+            var div_title = document.createElement("div");
+            var a_element = document.createElement("a");
+            a_element.href = link;
+            a_element.innerHTML = title;
+            div_title.appendChild(a_element);
+
+            var div_content = document.createElement("div");
+            div_content.innerHTML = content;
+
+            return div_title.outerHTML + div_content.outerHTML;
+        };
```

Comment:
I took a quick look at this code and it looks correct to me, but I'm not 100% sure. I don't know a good way to try this PR and QA the code. I think we don't have tests for this, either.

Probably it is worth having a second pair of eyes on the review checking the translation from the template used to this code.

Also, it would be good to have a longer comment explaining what this does exactly and examples of the HTML code generated by this.